tests/data/test_stream.py

```diff
@@ -0,0 +1,16 @@
+import pandas as pd
+import time
+import streamz
+from skmultiflow.data import from_generator, DataGenerator
+
+def test_sink_to_list():
+    df = pd.DataFrame(pd.np.random.randint(0, 100,size=(300, 4)), columns=list('ABCD'))
+    gen = DataGenerator(df)
+    stream = streamz.Source.from_generator(gen, batch_size=20, poll_interval=.001)
+
+    sub_df_list = stream.sink_to_list()
+    stream.start()
+    time.sleep(1.0)  # sleep long enough to let all sub_df pass through the stream
```

Comment:
Should certainly use `wait_for` here, or (perhaps and) run this as a [coroutine](https://github.com/python-streamz/streamz/blob/master/streamz/utils_test.py#L57) and use `await_for`. You can't guarantee that CI runs as fast as your local test system, and you don't want lots of `sleep` statements to slow down the testing process. Of course, if the simple data generators are not aync at all, then you will need no wait.

Reply:
Thanks for that precious comment
I'll update this

Reply:
@martindurant actually I have some problems with `wait_for`

`sub_df_list` is empty when it's declared
What I usually do is I make a call to pd.concat on this list
calling pd.concat on the current state of the list raises an error, even using `wait_for`, which is normal since the concat is done before the waiting ...

Reply:
I managed to to this by creating a custom predicate and submit it to wait_for